x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -584,6 +586,24 @@ private static boolean checkGroupMatch(Expression e, Node<?> source, List<Expres
         }
         return false;
     }
+    
+    private static void checkHistogramInGrouping(LogicalPlan p, Set<Failure> localFailures) {
+        // check if the query has a grouping function (Histogram) but no GROUP BY
+        if (p instanceof Project) {
+            Project proj = (Project) p;
+            proj.projections().forEach(e -> e.forEachDown(f -> localFailures.add(fail(f, "[%s] needs to be part of the grouping",
+                    Expressions.name(f))), GroupingFunction.class));
+        } else if (p instanceof Aggregate) {
+            // if it does have a GROUP BY, check if the groupings contain the grouping functions (Histograms) 
+            Aggregate a = (Aggregate) p;
+            a.aggregates().forEach(agg -> agg.forEachDown(e -> {
+                if (Expressions.anyMatch(a.groupings(), g -> {return g instanceof Function && e.functionEquals((Function) g);}) == false
```

Comment:
`anyMatch(a.groupings(), g -> g instanceof Function && functionEquals((Function) g)`
Avoids the use of `{`, `return` and `;`
Also like above it will be readable if you put each statement on a new line